### PR TITLE
JCLOUDS-1629: Upgrade to Guice 7.0.0

### PR DESCRIPTION
**Pull request by gaul**
This also changes from javax to jakarta annotations.  @basil

### CODE REVIEW

File: project/pom.xml [hunk 1]
```diff
@@ -223,7 +223,7 @@
     <!-- General dependency versions -->
     <gson.version>2.10.1</gson.version>
     <guava.version>32.0.0-jre</guava.version>
-    <guice.version>5.1.0</guice.version>
+    <guice.version>7.0.0</guice.version>
```
Comment (basil) [inline project/pom.xml:226]:
```suggestion
    <guice.version>6.0.0</guice.version>
```

**Review by basil — COMMENTED** [review]
I confirmed that Jenkins Artifact Manager on S3 plugin tests pass with this PR with both Guice 6 and Guice 7. However, I think it is too aggressive for JClouds to depend on Guice 7, as this will force all consumers of JClouds to upgrade to Guice 7 and therefore migrate from `javax` to `jakarta` annotations. I think it is preferable for JClouds to depend on Guice 6, which supports both `javax` and `jakarta` annotations, but to use `jakarta` annotations itself. That way, JClouds will be forward compatible with Guice 7 without forcing all consumers to migrate.

File: project/pom.xml (hunk 1 above)
Comment (gaul) [reply project/pom.xml:226]: You are welcome to investigate the failures with 6.0.0 but I don't have the time or the interest:

```
 jclouds-compute: Compilation failure
[ERROR] /home/gaul/work/jclouds/compute/src/main/java/org/jclouds/compute/config/BaseComputeServiceContextModule.java:[9
2,71] no suitable method found for toProvider(java.lang.Class<org.jclouds.compute.config.GetLoginForProviderFromProperti
esAndStoreCredentialsOrReturnNull>)
[ERROR]     method com.google.inject.binder.LinkedBindingBuilder.toProvider(com.google.inject.Provider<? extends org.jcl
ouds.domain.LoginCredentials>) is not applicable
[ERROR]       (argument mismatch; java.lang.Class<org.jclouds.compute.config.GetLoginForProviderFromPropertiesAndStoreCr
edentialsOrReturnNull> cannot be converted to com.google.inject.Provider<? extends org.jclouds.domain.LoginCredentials>)
[ERROR]     method com.google.inject.binder.LinkedBindingBuilder.toProvider(javax.inject.Provider<? extends org.jclouds.
domain.LoginCredentials>) is not applicable
[ERROR]       (argument mismatch; java.lang.Class<org.jclouds.compute.config.GetLoginForProviderFromPropertiesAndStoreCredentialsOrReturnNull> cannot be converted to javax.inject.Provider<? extends org.jclouds.domain.LoginCredentials>)
[ERROR]     method com.google.inject.binder.LinkedBindingBuilder.toProvider(java.lang.Class<? extends javax.inject.Provider<? extends org.jclouds.domain.LoginCredentials>>) is not applicable
[ERROR]       (argument mismatch; java.lang.Class<org.jclouds.compute.config.GetLoginForProviderFromPropertiesAndStoreCredentialsOrReturnNull> cannot be converted to java.lang.Class<? extends javax.inject.Provider<? extends org.jclouds.domain.LoginCredentials>>)
[ERROR]     method com.google.inject.binder.LinkedBindingBuilder.toProvider(com.google.inject.TypeLiteral<? extends javax.inject.Provider<? extends org.jclouds.domain.LoginCredentials>>) is not applicable
[ERROR]       (argument mismatch; java.lang.Class<org.jclouds.compute.config.GetLoginForProviderFromPropertiesAndStoreCredentialsOrReturnNull> cannot be converted to com.google.inject.TypeLiteral<? extends javax.inject.Provider<? extends org.jclouds.domain.LoginCredentials>>)
[ERROR]     method com.google.inject.binder.LinkedBindingBuilder.toProvider(com.google.inject.Key<? extends javax.inject.Provider<? extends org.jclouds.domain.LoginCredentials>>) is not applicable
[ERROR]       (argument mismatch; java.lang.Class<org.jclouds.compute.config.GetLoginForProviderFromPropertiesAndStoreCredentialsOrReturnNull> cannot be converted to com.google.inject.Key<? extends javax.inject.Provider<? extends org.jclouds.domain.LoginCredentials>>)
```

Comment (basil) [reply project/pom.xml:226]: > I don't have the time or the interest

Neither do I.

**Pull request merged**